### ISSUE TITLE
[Feature Request] Ability to open threads from any reply in that thread

### ISSUE THREAD

**Issue by theolegest** [telegramdesktop/tdesktop#16784]
### Is your feature request related to a problem?
Right now the only way to view a thread in a chat is to select "Show X replies" option of the original message. That message can be very far in chat history and hard to access.

### Describe the solution you'd like
Adding a "View thread" context option for any message that belongs to a thread eliminates that problem. It would also be consistent with the mobile version.

**Comment by RememberTheAir** [comment]
The mobile version behaves the same: the "view thread" option is shown only in supergroups that are linked to a channel

**Comment by Aokromes** [comment]
so, api?

**Comment by RememberTheAir** [comment]
I don't think so. But for sure it's something that should work the same way on every platform. [This card](https://bugs.telegram.org/c/1087) come close to what OP is asking for, but it's not quite it. Maybe a ticket there should be opened

**Comment by theolegest** [comment]
> The mobile version and tdesktop behave the same: the "view thread" option is shown only in supergroups that are linked to a channel

Is that the case? I tried to open both context menus from mobile and desktop and there was no "view thread" option on desktop. I test both Windows and Linux version

![message desktop](https://user-images.githubusercontent.com/34276758/128734081-500cf9be-af6d-4ba3-bd5b-1650a9ae4db9.jpeg)

![reply desktop](https://user-images.githubusercontent.com/34276758/128734151-5dc794e9-b834-4b3a-9ed2-7ebb4f4fe0a0.jpeg)

![message mobile](https://user-images.githubusercontent.com/34276758/128734170-92307da1-b694-4583-9fd4-6ff6d3b82d5a.jpeg)

![reply mobile](https://user-images.githubusercontent.com/34276758/128734191-2d7d18e9-7a25-4c22-a2cc-0b3e2d732c3c.jpeg)

**Comment by RememberTheAir** [comment]
Uh, maybe the iOS app behaves differently? Just tried from android and the reply doesn't show the "view thread" option

**Comment by gonhidi** [comment]
> Uh, maybe the iOS app behaves differently? Just tried from android and the reply doesn't show the "view thread" option

The Android client gained the functionality some time ago, then lost it for a few months (in August or earlier), but it has been back since mid-October.

**Comment by RememberTheAir** [comment]
I tried this on mobile in a public group without a linked channel, and I still can't open the replies thread from a reply (there is no option to open the thread in the message menu). I can only open it from the root message 🤔

**Comment by gonhidi** [comment]
You are right that the “view thread” option is only available for some groups. I though that its disappearance from the Android client while your conversation took place might somehow explain why you were getting different results.

Going further back, you noted that both the mobile and desktop versions offer the same option. In my case, however, for those groups where there is a “view thread” option for replies in the current Android client, I have never managed to see the equivalent functionality in macOS's native clients (currently, neither on Telegram Desktop 3.4.3 — the one this repository is about — nor on Telegram 8.4.1 226307 Beta, both on macOS Catalina 10.15.7); in those, I can only view the thread using the “view replies” option on the first message of a conversation. Were you using a different operating system or desktop client?

**Comment by RememberTheAir** [comment]
@gonhidi you are right, tdesktop doesn't shown any "view thread" option even when it should (that is, in supergroups with a linked channel where the message belongs to a thread). Tested on 3.4.8

**Comment by RememberTheAir** [comment]
Shouldn't this be brought to https://suggestions.telegram.org? Since the behavior is established and should be consistent everywhere, this is not a strictly client-side enhancement
